usaspending_api/common/query_with_filters.py

```diff
@@ -830,7 +919,11 @@ def generate_awards_elasticsearch_query(cls, filters: dict, **options) -> ES_Q:
 
     @classmethod
     def generate_subawards_elasticsearch_query(cls, filters: dict, **options) -> ES_Q:
-        return cls._generate_elasticsearch_query(filters, _QueryType.SUBAWARDS, **options)
+        cls.filter_lookup[_Keywords.underscore_name] = _SubawardsKeywords
+        cls.filter_lookup[_SubawardsPrimeSubAwardTypes.underscore_name] = _SubawardsPrimeSubAwardTypes
+        resp = cls._generate_elasticsearch_query(filters, _QueryType.SUBAWARDS, **options)
+        cls.filter_lookup[_Keywords.underscore_name] = _Keywords
```

Comment:
Is it safe to temporarily override this map attribute at the class level? I see you are setting it back to `_Keywords` at the end, but could these situations occur:
- `generate_awards_elasticsearch_query`, `generate_transactions_elasticsearch_query` or `generate_accounts_elasticsearch_query` is called while `generate_subawards_elasticsearch_query` is running
- A failure occurs in `cls._generate_elasticsearch_query(filters, _QueryType.SUBAWARDS, **options)`, which would mean that it never gets reset back? 

On another note, since `cls.filter_lookup[_Keywords.underscore_name] = _SubawardsKeywords`is not being removed from the map at the of the function, would it make sense to go ahead and include it in the main `filter_lookup` definition. 

With these in mind, it might be a good idea to add some checks to these two new types of Subaward-specific filters to make sure QueryType is Subawards

Reply:
> Is it safe to temporarily override this map attribute at the class level?

Yes I checked this. In what case would generate_awards_elasticsearch_query, generate_transactions_elasticsearch_query or generate_accounts_elasticsearch_query be called while generate_subawards_elasticsearch_query is running? Our application is single threaded. And in jenkins any async calls each have their own python interprettor, so class attributes are not shared.

Reply:
> On another note, since cls.filter_lookup[_Keywords.underscore_name] = _SubawardsKeywordsis not being removed from the map at the of the function, would it make sense to go ahead and include it in the main filter_lookup definition.

No doesn't make sense, no reason to.

Reply:
> Yes I checked this. In what case would generate_awards_elasticsearch_query, generate_transactions_elasticsearch_query or generate_accounts_elasticsearch_query be called while generate_subawards_elasticsearch_query is running? Our application is single threaded. And in jenkins any async calls result in two seperate Python processes that have their own python interpreter, so class attributes are not shared.

Sounds good. I wasn't sure about this, so I just wanted to make sure that it was considered

Reply:
> No doesn't make sense, what's the concern here?

I think we should deviate from base cases (in this case defining our filters in `filter_lookup`) as little as possible and only if there is a specific reason. In this case, it's going to permanently end up in `filter_lookup` after `generate_subawards_elasticsearch_query()` is run for the first time, which indicates that it is safe to stay there. In that case, why not follow the base case and define it with the other filters?

Reply:
Other filters do not need it. That's why. I can make the change, however I need a better reason beyond, it's always been done this way. 

Overall, I'm not too happy with the design pattern of this class anyway.  The way the class is, is not modular and is hard to extend. This subawards work showed me this because we are having to create new filter queries in a hacky way instead of extending and overwriting existing behaviors. That's why I implemented how I did.

Reply:
If you feel it's important queries shouldn't have access to it, then it should be removed from `cls.filter_lookup` after `cls._generate_elasticsearch_query` is run, similar to how the keywords filter is reset. 

I'm okay with whatever you decide on the items in this thread. Just trying to provide some feedback. The solid request I'd like to make is to please add comments to these new filters, including `_TransactionKeywordSearch` explaining what `_QueryTypes` they each are or are not compatible with.

Reply:
Thank you!